src/push/index.ts

```diff
@@ -72,6 +72,9 @@ export async function push(monitors: Monitor[], options: PushOptions) {
       throw formatAPIError(statusCode, error, message);
     }
     body.setEncoding('utf-8');
+    body.on('error', err => {
```

Comment:
IMO, This should not be needed, when we consume the stream, any error should be propagate and should blow up inside try-catch, can you give the test case where we need this?

Reply:
And the way we are treating this error is not the correct way. In anycase, we need a test case for this.

Reply:
++ - I tested this again and with this code the result is that for an error like the socket closing, we log it and then see it logged from the catch again. So yes, I think we should remove this.

My main concern here was we are providing a call to the RXJS `Subject.error` in the Kibana side. This presumably would send a message for something like a Saved Object failure, or some other strange situation. We seem not to handle that case (i.e. connection did not fail but some other error happened and was sent over the stream). Maybe this is a knowledge gap on my part as it relates to streams, but that was why I was trying this. I can't actually create an this state on purpose in my testing, so I think it's ok if we just remove this and rely on the `try...catch`.

Reply:
So in the RX side, subject.error or subject.complete would actually end the connection and close the stream with the final bytes. In theory, it would need to call `stream.emit("close")` or `stream.emit("error")`. In both cases we would catch it as the iteration would need to consume the body and end up throwing.

Reply:
Thanks for the explanation 👍